src/main/kotlin/uk/gov/justice/digital/assessments/services/AssessmentUpdateService.kt

```diff
@@ -67,7 +68,15 @@ class AssessmentUpdateService(
     assessmentRepository.save(episode.assessment)
     log.info("Saved episode ${episode.episodeUuid} for assessment ${episode.assessment?.assessmentUuid}")
 
-    return AssessmentEpisodeDto.from(episode, oasysResult)
+    val updatedEpisode = AssessmentEpisodeDto.from(episode, oasysResult)
```

Comment:
we could add the predictors as a parameter of this from builder and avoid mutating the updatedEpisode on line 77

Reply:
Had to move a few things around but I think I've managed it 😄